lib/ansible/modules/cloud/azure/azure_rm_route.py

```diff
@@ -0,0 +1,215 @@
+#!/usr/bin/python
+#
+# Copyright (c) 2018 Yuwei Zhou, <yuwzho@microsoft.com>
+#
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+ANSIBLE_METADATA = {'metadata_version': '1.1',
+                    'status': ['preview'],
+                    'supported_by': 'community'}
+
+
+DOCUMENTATION = '''
+---
+module: azure_rm_route
+version_added: "2.7"
+short_description: Manage Azure route resource.
+description:
+    - Create, update or delete a route.
+options:
+    resource_group:
+        description:
+            - name of resource group.
+        required: true
+    name:
+        description:
+            - name of the route.
+        required: true
+    state:
+        description:
+            - Assert the state of the route. Use 'present' to create or update and
+              'absent' to delete.
+        default: present
+        choices:
+            - absent
+            - present
+    address_prefix:
+        description:
+            - The destination CIDR to which the route applies.
+    next_hop_type:
+        description:
+            - The type of Azure hop the packet should be sent to.
+        choices:
+            - virtual_network_gateway
+            - vnet_local
+            - internet
+            - virtual_appliance
+            - none
+        default: 'none'
+    next_hop_ip_address:
+        description:
+            - The IP address packets should be forwarded to.
+            - Next hop values are only allowed in routes where the next hop type is VirtualAppliance.
+    route_table:
```

Comment:
Should probably be `route_table_name`? Unless we want to do what we've done in some other places with resource references, allowing a name/url/dict (in which case resource_group as a top-level arg might not be necessary here)?